### DISCUSSION TITLE
Custom integration?

### DISCUSSION THREAD

**Discussion by jappish84** [gllmlbrt/Alsavo-HassOS#1] [General]
Thanks for the work and clear instructions on using this repo. I'm trying to set this up for a friend but am struggling since he doesn't run hassos, but the supervised version on a rpi 3. He has another rpi 3 running dietpi, but there I'm struggling to build the repo as per the instructions. I'll try and build it on the dietpi system again. I found another docker build, but that's x86 :/

Any plans on making this a custom integration for HA so that it can run on any HA instance?

**Comment by gllmlbrt** [comment]
Thanks for the comment. Indeed the ideal solution would be a custom integration straight out of HACS, available on any Hass instance. Also skipping the whole MQTT templating thi ng and convoluted NodeRed.
But I am not even close to have the knowledge to put together a custom integration.

As a start anyway it would require to rewrite the AlsavoCtrl script from CPP to python. Which I don't feel capable of. Should anyone be up for that with proper python skills I would be happy to support with whatever I can.
